src/api/procedures/redeemToken.ts

```diff
@@ -41,21 +41,23 @@ export async function prepareRedeemToken(
 
   const portfolioBalance = await defaultPortfolio.getTokenBalances({ tokens: [ticker] });
 
-  if (portfolioBalance[0].total.lt(balance)) {
+  const { total, locked } = portfolioBalance[0];
+
+  if (total.minus(locked).lt(amount)) {
     throw new PolymeshError({
       code: ErrorCode.ValidationError,
       message: 'Insufficient balance',
     });
   }
 
-  if (!isDivisible) {
+  if (amount.mod(1) && !isDivisible) {
```

Comment:
`amount.mod(1)` will always be a `BigNumber`, and as such, always be truthy. Use `amount.decimalPlaces`, it is much more readable